include/yaclib/async/when_any.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
```

Comment:
Very useful pattern:
 Run async operation
 Run same operation, but different context, for example other database instance

 WhenAny(f1, f2).Then(...). ...

In this case, we need the first success, but not the first error. I mean any error is suitable for us, but if we return an error, it should be means that all futures have completed with an error.